edb/server/protocol/binary.pyx

```diff
@@ -499,6 +499,27 @@ cdef class EdgeConnection(frontend.FrontendConnection):
         else:
             return edgeql.NormalizedSource.from_string(text)
 
+    async def _suppress_tx_timeout(self):
+        async with self.with_pgcon() as conn:
+            await conn.sql_execute(b'''
+                select pg_catalog.set_config(
+                    'idle_in_transaction_session_timeout', '0', true)
+            ''')
+
+    async def _restore_tx_timeout(self, dbview.DatabaseConnectionView dbv):
+        old_timeout = dbv.get_session_config().get(
+            'session_idle_transaction_timeout',
+        )
+        timeout = (
+            'NULL' if not old_timeout
+            else repr(old_timeout.value.to_backend_str())
+        )
+        async with self.with_pgcon() as conn:
+            await conn.sql_execute(f'''
+                select pg_catalog.set_config(
+                    'idle_in_transaction_session_timeout', {timeout}, false)
```

Comment:
Should probably also be `true` here for symmetry.

Reply:
Hm, though that setting is *not* transaction local. It should persist when the transaction is committed. It looks like that does still work if it is true (because if it was configured earlier in the transaction, that config wasn't local), but that seems more subtle